### PR TITLE
Stop benchmark as soon as all transactions were covered by blocks

### PR DESCRIPTION
**Pull request by karolk91**

### CODE REVIEW

File: src/stats.rs [hunk 1]
```diff
@@ -1,41 +1,75 @@
...
 where
     F: Fn(sp_core::H256) -> Fut + Copy,
     Fut: Future<Output = color_eyre::Result<Vec<sp_core::H256>>>,
 {
-    block_stats
+    let block_stats_arc = Arc::new(Mutex::new(block_stats));
+    let remaining_hashes_arc = Arc::new(Mutex::new(remaining_hashes));
+
+    let remaining_hashes = remaining_hashes_arc.clone();
+    let stream = StreamNoOpUntil {
```
Comment (ascjones) [inline src/stats.rs:52]: Would using https://docs.rs/futures/latest/futures/stream/fn.poll_fn.html work here to save creating `StreamNoOpUntill`?

**Review by ascjones — COMMENTED** [review]
Conflict needs resolving.

File: src/stats.rs (hunk 1 above)
Comment (karolk91) [reply src/stats.rs:52]: looks promising, I will now try to use it

Comment (karolk91) [reply src/stats.rs]: @ascjones , just changed the implementation to use poll_fn instead, thank you! I had been looking for a simpler solution

**Pull request merged**